bin/pre-commit.sh

```diff
@@ -1,37 +1,23 @@
 #!/bin/bash
 
-if [ -n "$failIt" ]; then
-  echo "Expecting failIt to be empty."
-  exit 1
-fi
+rc=0
 
-wantEmpty=$(gofmt -s -d -l . 2>&1 )
-if [ -n "$wantEmpty" ]; then
-  printf >&2 'gofmt failed for:\n%s\n' "$wantEmpty"
-  failIt=1
-fi
+go_dirs=$(go list -f {{.Dir}} ./...)
```

Comment:
- We're in the habit of using [shellcheck](https://www.shellcheck.net/) to lint our bash scripts. We'd like to quote `"{{.Dir}}"` rather than `{{.Dir}}` to [make our shellcheck editor plugin happy](https://github.com/koalaman/shellcheck/wiki/SC1083) :)
- We'd like to check the list of returned directories. If someone were to introduce a directory with whitespace in the directory name, then the tests below would fail with potentially mysterious error messages. We can check for this early, with something like: 
```sh
go_dirs=$(go list -f "{{.Dir}}" ./...)

for dir in $go_dirs
do
  [[ -d $dir ]] || {
    echo "Problem with directory $dir" >&2
    exit 1
  }
done
```

Reply:
Cool! Maybe we should run shellcheck as part of this script :-)

I'm glad you caught that. I was very aware of this though, just too lazy to fix it. Thank you for forcing me to do it :D 
I'll try to find a better fix though, we can probably do better with `xargs -0`.

Reply:
The question also is: Is it allowed to have whitespaces in directories/filenames, which means: is it valid to have whitespaces in package names?

Reply:
The problem is that `go list -f "{{.Dir}}"` returns the absolute path for each directory. Which means that if your code lives in "/home/my**SP**username/my-go-path/mypackagewithnospace", it's going to fail, but probably shouldn't